code/game/objects/items/weapons/implants/implant_explosive.dm

```diff
@@ -31,13 +31,44 @@
 	to_chat(imp_in, "<span class='danger'>You activate your microbomb bio-chip.</span>")
 //If the delay is short, just blow up already jeez
 	if(delay <= 7)
-		explosion(src, heavy, medium, weak, weak, flame_range = weak)
-		if(imp_in)
-			imp_in.gib()
-		qdel(src)
+		self_destruct()
 		return
 	timed_explosion()
 
+/// Gib the implantee and delete their destructible contents.
+/obj/item/implant/explosive/proc/self_destruct()
+	if(!imp_in)
+		return
+
+	explosion(src, heavy, medium, weak, weak, flame_range = weak)
+
+	// In case something happens to the implantee between now and the
+	// self-destruct
+	var/current_location = get_turf(imp_in)
+
+	var/destructed_items = list()
+
+	// Iterate over the implantee's contents and take out indestructible
+	// things to avoid having to worry about containers and recursion
+	for(var/obj/item/I in imp_in.get_contents())
+		if(I)
+			if(I == src) // Don't delete ourselves prematurely
+				continue
+			// Drop indestructible items on the ground first, to avoid them
+			// getting deleted when destroying the rest of the items, which we
+			// track in a list to qdel afterwards
+			if(I.resistance_flags & INDESTRUCTIBLE)
+				I.forceMove(current_location)
+			else
+				destructed_items += I
```

Comment:
```suggestion
		if(I == src) // Don't delete ourselves prematurely
			continue
		// Drop indestructible items on the ground first, to avoid them
		// getting deleted when destroying the rest of the items, which we
		// track in a list to qdel afterwards
		if(I.resistance_flags & INDESTRUCTIBLE)
			I.forceMove(current_location)
		else
			destructed_items += I
```
Can't be null if you do a filtered for loop

Reply:
Done